### PR TITLE
pythonPackages.neoteroi-mkdocs: init at 1.0.4

### PR DESCRIPTION
**Pull request by zimbatm**
## Description of changes

And a few dependencies.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by aldoborrero — APPROVED** [review]

File: pkgs/development/python-modules/essentials-openapi/default.nix [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+  propagatedBuildInputs = [
+    click
+    essentials
+    httpx
+    jinja2
+    markupsafe
+    rich
+    setuptools
+  ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/essentials-openapi/default.nix:48]:
```suggestion
    pyyaml
    essentials
    markupsafe
  ];

  passthru.optional-dependencies = {
    full = [ click jinja2 rich httpx ];
  };
```

File: pkgs/development/python-modules/essentials-openapi/default.nix [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    hash = "sha256-j0vEMNXZ9TrcFx8iIyTFQIwF49LEincLmnAt+qodYmA=";
+  };
+
+  nativeCheckInputs = [
+    flask
+    hatchling
+    pydantic
+    pytestCheckHook
+    pythonImportsCheckHook
+    pyyaml
```
Comment (dotlambda) [inline pkgs/development/python-modules/essentials-openapi/default.nix:37]:
```suggestion
    setuptools
```

File: pkgs/development/python-modules/essentials-openapi/default.nix [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    owner = "Neoteroi";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-j0vEMNXZ9TrcFx8iIyTFQIwF49LEincLmnAt+qodYmA=";
+  };
+
+  nativeCheckInputs = [
+    flask
+    hatchling
+    pydantic
```
Comment (dotlambda) [inline pkgs/development/python-modules/essentials-openapi/default.nix:34]:
```suggestion
  nativeBuildInputs = [
    hatchling
  ];

  nativeCheckInputs = [
    flask
    pydantic
```

File: pkgs/development/python-modules/essentials/default.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "Neoteroi";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-WMHjBVkeSoQ4Naj1U7Bg9j2hcoErH1dx00BPKiom9T4=";
+  };
+
+  buildInputs = [ setuptools ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/essentials/default.nix:21]:
```suggestion
  nativeBuildInputs = [ setuptools ];
```

File: pkgs/development/python-modules/essentials/default.nix [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    pythonImportsCheckHook
```
Comment (dotlambda) [inline pkgs/development/python-modules/essentials/default.nix:25]:
```suggestion
```

File: pkgs/development/python-modules/mkdocs-plugins/default.nix [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  };
+
+  buildInputs = [
+    setuptools
+    hatchling
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    pythonImportsCheckHook
```
Comment (dotlambda) [inline pkgs/development/python-modules/mkdocs-plugins/default.nix:36]:
```suggestion
```

File: pkgs/development/python-modules/mkdocs-plugins/default.nix [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    essentials-openapi
+    click
+    jinja2
+    httpx
+    mkdocs
+    rich
+  ];
+
+  disabledTests = [
+    "test_contribs" # checks against it's own git repository
```
Comment (dotlambda) [inline pkgs/development/python-modules/mkdocs-plugins/default.nix:50]:
```suggestion
    "test_contribs" # checks against its own git repository
```

File: pkgs/development/python-modules/mkdocs-plugins/default.nix [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+  buildInputs = [
+    setuptools
+    hatchling
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    pythonImportsCheckHook
+    flask
```
Comment (dotlambda) [inline pkgs/development/python-modules/mkdocs-plugins/default.nix:37]:
```suggestion
    flask
    setuptools
```

**Review by dotlambda — DISMISSED** [review]

**Comment by zimbatm** [comment]
thanks @dotlambda !

**Comment by dotlambda** [comment]
Please squash and use the `python311Packages.` prefix.

**Comment by zimbatm** [comment]
Fixed, although I don't get why the prefix is `python311Packages` since it's available for all python 3 packages.

**Comment by dotlambda** [comment]
> I don't get why the prefix is `python311Packages` since it's available for all python 3 packages.

If you put `python3Packages` then @ofborg won't build the package automatically.

File: pkgs/development/python-modules/essentials-openapi/default.nix [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+  setuptools
+}:
+buildPythonPackage rec {
+  pname = "essentials-openapi";
+  version = "1.0.7";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "Neoteroi";
+    repo = pname;
```
Comment (dotlambda) [inline pkgs/development/python-modules/essentials-openapi/default.nix:26]:
```suggestion
    repo = "essentials-openapi";
```
https://github.com/NixOS/nixpkgs/issues/277994

File: pkgs/development/python-modules/essentials/default.nix [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+  lib,
+}:
+buildPythonPackage rec {
+  pname = "essentials";
+  version = "1.1.5";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "Neoteroi";
+    repo = pname;
```
Comment (dotlambda) [inline pkgs/development/python-modules/essentials/default.nix:16]:
```suggestion
    repo = "essentials";
```

File: pkgs/development/python-modules/mkdocs-plugins/default.nix [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+  setuptools,
+}:
+buildPythonPackage rec {
+  pname = "mkdocs-plugins";
+  version = "1.0.4";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "Neoteroi";
+    repo = pname;
```
Comment (dotlambda) [inline pkgs/development/python-modules/mkdocs-plugins/default.nix:24]:
```suggestion
    repo = "mkdocs-plugins";
```

File: pkgs/development/python-modules/mkdocs-plugins/default.nix [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+  httpx,
+  jinja2,
+  mkdocs,
+  pytestCheckHook,
+  pythonImportsCheckHook,
+  rich,
+  setuptools,
+}:
+buildPythonPackage rec {
+  pname = "mkdocs-plugins";
```
Comment (dotlambda) [inline pkgs/development/python-modules/mkdocs-plugins/default.nix:18]:
```suggestion
  pname = "neoteroi-mkdocs";
```
https://pypi.org/project/mkdocs-plugins/ doesn't exist

File: pkgs/top-level/python-packages.nix [hunk 13]
```diff
@@ -7158,6 +7162,7 @@ self: super: with self; {
   mkdocs-material = callPackage ../development/python-modules/mkdocs-material { };
   mkdocs-material-extensions = callPackage ../development/python-modules/mkdocs-material/mkdocs-material-extensions.nix { };
   mkdocs-minify-plugin = callPackage ../development/python-modules/mkdocs-minify-plugin { };
+  mkdocs-plugins = callPackage ../development/python-modules/mkdocs-plugins { };
```
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix:7165]:
```suggestion
  neoteroi-mkdocs = callPackage ../development/python-modules/neoteroi-mkdocs { };
```

File: pkgs/development/python-modules/mkdocs-plugins/default.nix [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+  src = fetchFromGitHub {
+    owner = "Neoteroi";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-UyTlgKWdBWckI9sBL4GRQtgNHYpHpZlWVOdmdQ+7lss=";
+  };
+
+  buildInputs = [
+    setuptools
+    hatchling
```
Comment (dotlambda) [inline pkgs/development/python-modules/mkdocs-plugins/default.nix:31]: Do we really need both?

File: pkgs/development/python-modules/mkdocs-plugins/default.nix (hunk 12 above)
Comment (zimbatm) [reply pkgs/development/python-modules/mkdocs-plugins/default.nix]: Fixed and renamed the folder as well

File: pkgs/development/python-modules/essentials-openapi/default.nix [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+    hatchling
+  ];
+
+  nativeCheckInputs = [
+    flask
+    httpx
+    pydantic
+    pytestCheckHook
+    rich
+    setuptools
```
Comment (zimbatm) [inline pkgs/development/python-modules/essentials-openapi/default.nix:41]: I had to re-add httpx and rich as check dependencies

**Comment by zimbatm** [comment]
alright, I think it's ready now

File: pkgs/development/python-modules/essentials-openapi/default.nix (hunk 15 above)
Comment (dotlambda) [inline pkgs/development/python-modules/essentials-openapi/default.nix:41]:
```suggestion
```
According to https://github.com/Neoteroi/essentials-openapi/blob/4ffc7b1921835a22a386494c0acb4f4390f5f1a4/tests/common.py#L18, `pkg_resources` is only needed for Python 3.8.

File: pkgs/development/python-modules/neoteroi-mkdocs/default.nix [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+  };
+
+  buildInputs = [
+    hatchling
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    flask
+    setuptools
```
Comment (dotlambda) [inline pkgs/development/python-modules/neoteroi-mkdocs/default.nix:36]:
```suggestion
```
same here

File: pkgs/development/python-modules/neoteroi-mkdocs/default.nix [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+  lib,
+  click,
+  essentials-openapi,
+  flask,
+  hatchling,
+  httpx,
+  jinja2,
+  mkdocs,
+  pytestCheckHook,
+  pythonImportsCheckHook,
```
Comment (dotlambda) [inline pkgs/development/python-modules/neoteroi-mkdocs/default.nix:13]:
```suggestion
```

File: pkgs/development/python-modules/essentials/default.nix [hunk 18]
```diff
@@ -0,0 +1,38 @@
+{
+  buildPythonPackage,
+  fetchFromGitHub,
+  setuptools,
+  pytestCheckHook,
+  pythonImportsCheckHook,
```
Comment (dotlambda) [inline pkgs/development/python-modules/essentials/default.nix:6]:
```suggestion
```

File: pkgs/development/python-modules/essentials-openapi/default.nix [hunk 19]
```diff
@@ -0,0 +1,65 @@
...
+  click,
+  essentials,
+  flask,
+  hatchling,
+  httpx,
+  jinja2,
+  markupsafe,
+  pydantic,
+  pytestCheckHook,
+  pythonImportsCheckHook,
```
Comment (dotlambda) [inline pkgs/development/python-modules/essentials-openapi/default.nix:14]:
```suggestion
```

Comment (zimbatm) [reply pkgs/development/python-modules/essentials-openapi/default.nix]:
```
error: builder for '/nix/store/4annsxfapvaagljlz1vq7iawza6r1md5-python3.11-essentials-openapi-1.0.7.drv' failed with exit code 2;
       last 25 log lines:
       >     return _bootstrap._gcd_import(name[level:], package, level)
       > tests/test_v2.py:36: in <module>
       >     from tests.common import debug_result
       > tests/common.py:6: in <module>
       >     import pkg_resources
       > E   ModuleNotFoundError: No module named 'pkg_resources'
       > ______________________ ERROR collecting tests/test_v3.py _______________________
       > ImportError while importing test module '/build/source/tests/test_v3.py'.
       > Hint: make sure your test modules/packages have valid Python names.
       > Traceback:
       > /nix/store/y027d3bvlaizbri04c1bzh28hqd6lj01-python3-3.11.7/lib/python3.11/importlib/__init__.py:126: in import_module
       >     return _bootstrap._gcd_import(name[level:], package, level)
       > tests/test_v3.py:46: in <module>
       >     from tests.common import debug_result
       > tests/common.py:6: in <module>
       >     import pkg_resources
       > E   ModuleNotFoundError: No module named 'pkg_resources'
       > =========================== short test summary info ============================
       > ERROR tests/test_cli.py
       > ERROR tests/test_mk_v3.py
       > ERROR tests/test_v2.py
       > ERROR tests/test_v3.py
       > !!!!!!!!!!!!!!!!!!! Interrupted: 4 errors during collection !!!!!!!!!!!!!!!!!!!!
       > ============================== 4 errors in 0.25s ===============================
       > /nix/store/yq9qag5ak3w15nhcqplnhxpny6295l36-stdenv-linux/setup: line 1582: pop_var_context: head of shell_variables not a function context
```

File: pkgs/development/python-modules/essentials-openapi/default.nix (hunk 15 above)
Comment (zimbatm) [reply pkgs/development/python-modules/essentials-openapi/default.nix]:
```
error: builder for '/nix/store/4annsxfapvaagljlz1vq7iawza6r1md5-python3.11-essentials-openapi-1.0.7.drv' failed with exit code 2;
       last 25 log lines:
       >     return _bootstrap._gcd_import(name[level:], package, level)
       > tests/test_v2.py:36: in <module>
       >     from tests.common import debug_result
       > tests/common.py:6: in <module>
       >     import pkg_resources
       > E   ModuleNotFoundError: No module named 'pkg_resources'
       > ______________________ ERROR collecting tests/test_v3.py _______________________
       > ImportError while importing test module '/build/source/tests/test_v3.py'.
       > Hint: make sure your test modules/packages have valid Python names.
       > Traceback:
       > /nix/store/y027d3bvlaizbri04c1bzh28hqd6lj01-python3-3.11.7/lib/python3.11/importlib/__init__.py:126: in import_module
       >     return _bootstrap._gcd_import(name[level:], package, level)
       > tests/test_v3.py:46: in <module>
       >     from tests.common import debug_result
       > tests/common.py:6: in <module>
       >     import pkg_resources
       > E   ModuleNotFoundError: No module named 'pkg_resources'
       > =========================== short test summary info ============================
       > ERROR tests/test_cli.py
       > ERROR tests/test_mk_v3.py
       > ERROR tests/test_v2.py
       > ERROR tests/test_v3.py
       > !!!!!!!!!!!!!!!!!!! Interrupted: 4 errors during collection !!!!!!!!!!!!!!!!!!!!
       > ============================== 4 errors in 0.25s ===============================
       > /nix/store/yq9qag5ak3w15nhcqplnhxpny6295l36-stdenv-linux/setup: line 1582: pop_var_context: head of shell_variables not a function context
```

**Comment by zimbatm** [comment]
ok, the last round of review hit past the threshold of diminishing return and started breaking things. I think the changes are fine now and will be merging them soon.

**Pull request merged**